iohk-monitoring/src/Cardano/BM/Data/Counter.lhs

```diff
@@ -69,7 +69,7 @@ nameCounter (Counter RTSStats           _ _) = "RTS"
 \subsubsection{CounterState}\label{code:CounterState}\index{CounterState}
 \begin{code}
 data CounterState = CounterState {
-      csCounters   :: [Counter]
+      csCounters   :: ![Counter]
```

Comment:
Note that this just force the list to WHNF. Using `_` to indicate a thunk, this changes `csCounters = _` into `csCounters = _ : _`, so this is not forcing all the elements of the list.

Reply:
reverted it.

Reply:
If this really is leaking space (we need some evidence for that first), then removing the bang is not the solution. Either use a stricter data structure than a list or force its contents (e.g., `Cardano.Prelude.forceElemsToWHNF`).